### ISSUE TITLE
 'Microsoft.WindowsDesktop.App', version '3.1.0' was not found

### ISSUE THREAD

**Issue by dev-ptdi** [emgucv/emgucv#339]
Hi All,

I deploy emgu.cv on CentOS 7, I got  The framework 'Microsoft.WindowsDesktop.App', version '3.1.0' was not found ..

below what I have:

[root@demo Microsoft.AspNetCore.App]# dotnet --list-runtimes
Microsoft.AspNetCore.App 3.0.3 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.2 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.AspNetCore.App 3.1.5 [/usr/share/dotnet/shared/Microsoft.AspNetCore.App]
Microsoft.NETCore.App 3.0.3 [/usr/share/dotnet/shared/Microsoft.NETCore.App]

I am not using Microsoft.WindowsDesktop.App in my ASP Net Core project..

**Comment by emgucv** [comment]
Which Emgu CV version are you using?

Are you using nuget package from nuget.org?

**Comment by dev-ptdi** [comment]
Yes



We are happy to help,

Winanjaya
+62 817 145779
————————————————
Sent from my iPhone. 
My apologize for the typos & brevity. 
————————————————

**Comment by emgucv** [comment]
Is it version 4.3 or v4.2?

**Comment by dev-ptdi** [comment]
I use the latest 4.3



We are happy to help,

Winanjaya
+62 817 145779
————————————————
Sent from my iPhone. 
My apologize for the typos & brevity. 
————————————————

**Comment by sn4k3** [comment]
I have the same problem, when reference EmguCV from nuget in a .net core 3.1 project, that dependency will bring Windows dependency:

`"Emgu.CV/4.3.0.3890": {
        "type": "package",
        "dependencies": {
          "System.Drawing.Common": "4.7.0",
          "System.Drawing.Primitives": "4.3.0",
          "System.Runtime": "4.3.1",
          "System.Runtime.InteropServices.RuntimeInformation": "4.3.0"
        },
        "compile": {
          "lib/netcoreapp3.1/Emgu.CV.Platform.NetCore.dll": {},
          "lib/netcoreapp3.1/Emgu.CV.Platform.NetStandard.dll": {}
        },
        "runtime": {
          "lib/netcoreapp3.1/Emgu.CV.Platform.NetCore.dll": {},
          "lib/netcoreapp3.1/Emgu.CV.Platform.NetStandard.dll": {}
        },
        "frameworkReferences": [
          "Microsoft.WindowsDesktop.App.WindowsForms"
        ]
      },`

Forcing a runtime doesn't matter: dotnet build --runtime ubuntu.16.04-x64

**Comment by jonmiller1** [comment]
I'm stuck on this on this too.  Changing test console program to .NET Standard 2.0 didn't help either.  Any ideas?  Can someone submit a working sample project?

**Comment by sn4k3** [comment]
> I'm stuck on this on this too. Changing test console program to .NET Standard 2.0 didn't help either. Any ideas? Can someone submit a working sample project?

For me works if project is a ".NET Standard 2.0 Library", not test with ".NET Standard 2.0 Console" but with ".Net Core 3.1 Console" fails, so i guess is happen with consoles only?

**Comment by maartendweerdt** [comment]
Any update on this? I'm also stuck on this. Trying to run .NET Core 3.1 Console application on Raspberry pi. Any known workarounds yet (will try going to .NET Standard but would like to have it as a console app) or planned fixes for this?

**Comment by emgucv** [comment]
Fixed in the Git repository: 
The following dependency has been removed from the Emgu.CV nuget package
```
"frameworkReferences": [
"Microsoft.WindowsDesktop.App.WindowsForms"
]
```
We will have v4.4 out soon, which will include the above fix.

**Comment by d8ahazard** [comment]
> Fixed in the Git repository:
> The following dependency has been removed from the Emgu.CV nuget package
> 
> ```
> "frameworkReferences": [
> "Microsoft.WindowsDesktop.App.WindowsForms"
> ]
> ```
> 
> We will have v4.4 out soon, which will include the above fix.

Please make sure to update the linux-arm lib when you do. :D

**Comment by maartendweerdt** [comment]
> 
> Please make sure to update the linux-arm lib when you do. :D

Do you mean update the installation on my raspberry pi?

**Comment by d8ahazard** [comment]
> > Please make sure to update the linux-arm lib when you do. :D
> 
> Do you mean update the installation on my raspberry pi?

Kind of? In the past, the developer has uploaded prebuilt binaries to use on a pi. 

I literally *just* looked and saw that 4.3 had finally been released and added it to my app. This caused the above issue, which is promised to be fixed in 4.4. I was saying to the developers - "Pretty please create the 4.4 linux-arm binary when you release 4.4". :P

**Comment by emgucv** [comment]
The current release is 4.5.1 and the Emgu.CV nuget package targets .netstandard 2.0

Closing ticket now.
